sklearn/utils/_mask.py

```diff
@@ -1,19 +1,60 @@
 import numpy as np
+from scipy import sparse as sp
 
 from . import is_scalar_nan
 from .fixes import _object_dtype_isnan
 
 
-def _get_mask(X, value_to_mask):
-    """Compute the boolean mask X == value_to_mask."""
+def _get_dense_mask(X, value_to_mask):
     if is_scalar_nan(value_to_mask):
         if X.dtype.kind == "f":
-            return np.isnan(X)
+            Xt = np.isnan(X)
         elif X.dtype.kind in ("i", "u"):
             # can't have NaNs in integer array.
-            return np.zeros(X.shape, dtype=bool)
+            Xt = np.zeros(X.shape, dtype=bool)
         else:
             # np.isnan does not work on object dtypes.
-            return _object_dtype_isnan(X)
+            Xt = _object_dtype_isnan(X)
     else:
-        return X == value_to_mask
+        Xt = X == value_to_mask
+
+    return Xt
+
+
+def _get_mask(X, value_to_mask, sparse=False):
```

Comment:
From the usage of the `sparse` keyword parameter it seems like sparse is always equal to `issparse(X)`. If so, do we need the `sparse` keyword parameter?